lib/graphql/object/resolvers.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+# test_via: ../object.rb
+
+module GraphQL
+  class Object
+    module Resolvers
+      # Grab the Resolve strategy which was assigned at boot
+      class Metadata
+        def call(obj, args, ctx)
+          field_defn = ctx.field
+          resolve_strategy = field_defn.metadata[:resolve_strategy]
+          resolve_strategy.call(obj, args, ctx)
+        end
+      end
+
+      # This is assigned at build-time. It should be overridden during boot.
+      class Pending
+        ERROR_MESSAGE = "Can't resolve %{name} because its resolve strategy is still pending." +
+          " To resolve this error, call `Schema#boot` before running any queries."
+
+        def call(obj, args, ctx)
+          field_defn = ctx.field
+          owner_type = ctx.irep_node.owner_type
+          raise NotImplementedError, ERROR_MESSAGE % { name: "#{owner_type.name}.#{field_defn.name}" }
+        end
+      end
+
+      class Booted
+        def initialize(type_name:, field_name:, method_name:, object_class:)
+          @type_name = type_name
+          @field_name = field_name
+          @method_name = method_name
+          @object_class = object_class
+        end
+
+        def call(obj, args, ctx)
+          raise NotImplementedError, "Subclasses should implement a resolve function"
+        end
+      end
+
+      class MethodCall < Booted
+        def initialize(*)
+          super
+          @has_args = has_args?
+        end
+
+        def call(obj, args, ctx)
+          if @has_args
+            kwargs = {}
+            args.to_h.each { |k, v| kwargs[k.to_sym] = v }
+            obj.public_send(@method_name, kwargs)
```

Comment:
It would be pretty sweet to have some nicer ways of mapping the field's arguments to the method's arguments. Not sure if that's too magical though.

Mostly, I'm thinking automatically un-camelizing argument names and splatting them.

Reply:
:+1:  this splats them, so you can use keyword args, but it doesn't underscore-ize them yet 😊  definitely _should_ though.

Reply:
it doesn't _deep_ splat, that's definitely a todo, for input objects, i mean

Reply:
Splatting them is cooling. And the object being resolved against can be accessed in these methods just with `self`. But how is `ctx` accessible? Is there an accessor I'm missing?

Reply:
They're each available as `@object` and `@context`, but `self` is an instance of the `GraphQL::Object` subclass